File: Classroom/03.04/primesubtraction.py
# https://codeforces.com/problemset/problem/1238/A

cases = int(input())

for i in range(cases):
    a, b = map(int, input().split())
    
    if a % 2 == 0 and b % 2 == 0:
        print("YES")

    elif (a % 2 == 1 and b % 2 == 0) or (a % 2 == 0 and b % 2 == 1):
        if a - b >= 2:
            print("YES")
        else:
            print("NO")
    
    else:
        print("YES")

File: Classroom/04.03/primesubtraction.cpp
// https://codeforces.com/problemset/problem/1238/A

#include <bits/stdc++.h>

using namespace std;

#define ll long long

int main(){
    
    int cases;

    cin >> cases;

    for(int i = 0 ; i < cases ; i++){
        ll a, b;

        cin >> a >> b;

        a - b >= 2 ? cout << "YES" << "\n" : cout << "NO" << "\n";
    }
    
    return 0;
}